frontend/packages/ceph-storage-plugin/src/components/namespace-store/namespace-store-table.tsx

```diff
@@ -84,7 +84,7 @@ const NamespaceStoreTable: React.FC<NamespaceStoreTableProps> = (props) => {
   );
   const memoizedData = useDeepCompareMemoize(data, true);
   React.useEffect(() => {
-    if (!_.isEmpty(preSelected) && selectedRows.size === 0) {
+    if (!_.isEmpty(preSelected) && selectedRows.size === 0 && !_.isEmpty(memoizedData)) {
```

Comment:
I am a bit confused with with how this solves the problem.
memoizedData is empty when data is empty, it should not make any changes to pre-selected. 
and if data is passed, then memoizedData should have some values and this data is reflected in pre-selected.

It should work normally in those cases even without this change.

Can you elaborate a bit on this ?

Reply:
In this case, the data was empty initially but we had values in the `preSelected`. 
Thus
https://github.com/openshift/console/blob/b2d8c4bc7ee7b549242a02cbcc3617c8cd9f5fc5/frontend/packages/ceph-storage-plugin/src/components/namespace-store/namespace-store-table.tsx#L88 would return an empty array, leading that to overwriting the state `readNamespaceStore` with an empty array.

Reply:
After a while, we receive the data. But by that point our preselected values had already been overwritten.

Reply:
ack